### PR TITLE
feat(disruption): custom reporting

### PR DESCRIPTION
**Pull request by luphaz**
## What does this PR do?

- [x] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- This change aims to provide the ability to configure per disruption slack channel notification.

## Code Quality Checklist

- [x] The documentation is up to date.
- [x] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [x] I leveraged continuous integration testing
    - [x] by depending on existing `unit` tests or `end-to-end` tests.
    - [x] by adding new `unit` tests or `end-to-end` tests.
- [x] I manually tested the following steps:
    - [x] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

**Review by taihuynh167 — APPROVED** [review]
👍 nice work!

File: eventnotifier/slack/slack.go [hunk 1]
```diff
@@ -130,32 +131,21 @@ func (n *Notifier) Notify(dis v1beta1.Disruption, event corev1.Event, notifType
 		n.logger.Errorw("unable to retrieve disruption user info", "error", err, "disruption", dis.Name)
 	}
 
-	// To remove when we stop testing this feature
+	// Whenever a purpose is defined, we expect it to be available into all notifications sent messages
+	if nil != dis.Spec.Reporting && dis.Spec.Reporting.Purpose != "" {
+		disruptionBlocks = append(disruptionBlocks, slack.NewTextBlockObject("mrkdwn", "*Purpose:*\n"+dis.Spec.Reporting.Purpose, false, false))
+	}
+
 	if n.config.MirrorSlackChannelID != "" {
```
Comment (clairecng) [inline eventnotifier/slack/slack.go:139]: Note: I think we can remove the config.MirrorSlackChannelID safely now, we can do it in a separate PR if you think it's irrelevant here

**Review by clairecng — APPROVED** [review]
LGTM! 👍

File: eventnotifier/slack/slack.go (hunk 1 above)
Comment (luphaz) [reply eventnotifier/slack/slack.go:139]: I indeed saw `// To remove when we stop testing this feature` and wondered if we still wanted to remove this in the end 😉 

Let me know if we should, I will adapt this PR accordingly.

We may not be interested of being spammed of notifications into our channel so if we believe it's more annoying than useful we should remove this.

Why was it introduced in the first place?

Comment (luphaz) [reply eventnotifier/slack/slack.go:139]: Small bump here @clairecng 🤗 have you any idea on the question above?

Comment (clairecng) [reply eventnotifier/slack/slack.go:139]: @luphaz It was temporarily introduced for us to see the messages users would get when we first created the event observer, so that we could detect potentially wrong events sent. I believe we can remove it now.

**Review by taihuynh167 — APPROVED** [review]

File: eventnotifier/slack/slack.go (hunk 1 above)
Comment (luphaz) [reply eventnotifier/slack/slack.go:139]: I will do the removal in another PR just to confirm everything is OK on the schedule side first.
